### ISSUE TITLE
Ordered blocks change order after being modified

### ISSUE THREAD

**Issue by LZylstra** [directus/directus#24163]
### Describe the Bug

I have a Pages collection that has a Blocks collection, when I edit one of the blocks it send the one I changed to the bottom instead of keeping the order I had them in. [Video of it happening](https://www.loom.com/share/229ddbff46d14bed955d501d22e2a518?sid=6bd09e92-ac8a-459c-b545-e57f9ca67fb8)

### To Reproduce

Preconditions:
A parent collection with an ordered child collection

Steps:
1. Open the parent collection
2. Open one of the children and remember the order it was in previously
3. Make any change and save
4. Notice the order now that a change has been saved

### Directus Version

v11.3.2

### Hosting Strategy

Directus Cloud

**Comment by rijkvanzanten** [comment]
Based on the pages/blocks combo, I'm assuming you're using a M2A builder here? In that case, do you have a `sort` field configured for that relation?

**Comment by hanneskuettner** [comment]
This seems to be a direct result of this PR https://github.com/directus/directus/pull/20271 that was merged recently. Since the item drawer only returns edits and the sort field is not edited it is undefined in `applySort` and gets assigned a new sort value.

**Comment by LZylstra** [comment]
Yes I'm using the M2A builder and there is a sort field configured

**Comment by osmandvc** [comment]
As @hanneskuettner  rightfully mentioned, the problem occurred because the sort field wasn't being preserved in the edit operation, causing `applySort` to overwrite the already existing sorting field. The fix was to carry over `sortField` (if already existing) and set it in the editing object. Opened PR #24172 for it. 😊

Working example:

https://github.com/user-attachments/assets/3b8c7d97-dd0b-4f91-a631-e2a336ff2e3e

**Comment by abdonrd** [comment]
I have the same issue with M2M fields.

**Comment by osmandvc** [comment]
> I have the same issue with M2M fields.

I also realized that shortly after my fix, so I restructured `getItemEdits` to consistently preserve the sort field. The field is now captured before any other edits and included in all return paths. This ensures the sort field is preserved while editing and regardless of the relation type; M2M/M2A or any other interface which uses the `useRelationMultiple` hook. I pushed the new changes to my existing PR, looking forward for feedback from the Team 👌

**Comment by LevelUpGit** [comment]
@osmandvc We have same issue when batch editing, when editing one by one it works as expected. 

https://github.com/user-attachments/assets/78f15965-66ff-45d5-bf4b-18f1c63fa9b4
